core/schema/src/main/java/org/hawkular/metrics/schema/SchemaService.java

```diff
@@ -50,56 +51,91 @@ public void run(Session session, String keyspace, boolean resetDB, int replicati
                 "keyspace", keyspace,
                 "reset", resetDB,
                 "session", session,
                 "replicationFactor", replicationFactor,
                 "logger", logger
         );
+        return vars;
+    }
+
+    private List<String> getTags() {
         // TODO Add logic to determine the version tags we need to pass
         // For now, I am just hard coding the version tag, but a more robust solution would be
         // to calculate the tags using the current version stored in the sys_config table
         // and the new version which we can extract from any of our JAR manifest files.
         List<String> tags = asList("0.15.x", "0.18.x", "0.19.x", "0.20.x", "0.21.x", "0.23.x", "0.26.x", "0.27.x",
                 "0.30.x");
-        URI script = getScript();
-        cassalog.execute(script, tags, vars);
-
-        if (updateVersion) {
-            updateVersion(session, keyspace, 0, 0);
-        }
+        return tags;
     }
 
     public void updateVersion(Session session, String keyspace, long delay, int maxRetries) {
         doVersionUpdate(session, keyspace, delay, maxRetries);
     }
 
+    public void updateSchemaVersionSession(Session session, String keyspace, long delay, int maxRetries) {
+        doSchemaVersionUpdate(session, keyspace, delay, maxRetries);
+    }
+
+    public String getSchemaVersion(Session session, String keyspace) {
+        URI script = getScript();
+        CassalogBuilder builder = new CassalogBuilder();
+        Cassalog cassalog = builder.withKeyspace(keyspace).withSession(session).build();
+        Map<String, ?> vars = this.getVars(session, keyspace, false, 1);
+        List<ChangeSet> changeSets = cassalog.load(script, this.getTags(), vars);
+        if (changeSets.size() > 0) {
+            return changeSets.get(changeSets.size() - 1).getVersion();
+        }
+        return "";
+    }
+
     private void doVersionUpdate(Session session, String keyspace, long delay, int maxRetries) {
-        String configId = "org.hawkular.metrics";
-        String configName = "version";
         String version = VersionUtil.getVersion();
+        writeVersion("version", version, session, keyspace, delay, maxRetries);
+    }
+
+    private void doSchemaVersionUpdate(Session session, String keyspace, long delay, int maxRetries) {
+        String version = this.getSchemaVersion(session, keyspace);
+        writeVersion("schema-version", version, session, keyspace, delay, maxRetries);
+    }
+
+    private void writeVersion(String versionType, String version, Session session, String keyspace, long delay, int
+            maxRetries) {
+        String configId = "org.hawkular.metrics";
         SimpleStatement updateVersion = new SimpleStatement(
-                "INSERT INTO sys_config (config_id, name, value) VALUES (?, ?, ?)", configId, configName, version)
+                "INSERT INTO sys_config (config_id, name, value) VALUES (?, ?, ?)", configId, versionType, version)
```

Comment:
Since this query will not get executed multiple times, we should make it a PreparedStatement.